rotkehlchen/chain/ethereum/modules/lido/decoder.py

```diff
@@ -0,0 +1,526 @@
+            all_logs: list['EvmTxReceiptLog'],  # pylint: disable=unused-argument
+    ) -> list['EvmEvent']:
+        """Post decoding function to map the ERC20 Lido tokens reception"""
+        """TODO: To be checked with Rotki team, I am adding post decoding because I see that ERC20
+        transfers are not available in decoded_events at the time of decode by address rules
+        from what I have seen in the code. I got this solution looking at compound rules"""
+        for event in decoded_events:
+            if (
+                (event.counterparty == CPT_LIDO_ETH and event.notes is not None and event.extra_data is not None) and  # noqa: E501
+                (
+                    (event.event_type == HistoryEventType.RECEIVE and event.event_subtype == HistoryEventSubType.RECEIVE_WRAPPED) or  # noqa: E501
+                    (event.event_type == HistoryEventType.SPEND and event.event_subtype == HistoryEventSubType.RETURN_WRAPPED)  # noqa: E501
+                )
+            ):
+                event.notes = event.notes.format(amount=event.balance.amount)  # set the amount  # noqa: E501
+                break
+        return decoded_events
+
+    def _decode_lido_transfer_shares_for_steth_wrap(
+            self, context: DecoderContext,
+    ) -> DecodingOutput:
+        """Decode steth wrapping in wsteth from TransferShares log event"""
+
+        user_address = hex_or_bytes_to_address(context.tx_log.topics[1])
+
+        #  we need to retrieve the transferred stETH amount deterministically.
+        #  expected log index for the transferred steth is -1
+        #  expected log index for the transferred wsteth is -3
+        expected_steth_transfer_index = context.tx_log.log_index - 1
+        expected_wsteth_transfer_index = context.tx_log.log_index - 3
+        paired_wsteth_ratio = None
+        sth_wrapped_amount = None
+        wsteth_received_amount = None
+
+        for tx_log in context.all_logs:
+            if (
+                tx_log.log_index == expected_steth_transfer_index and
+                tx_log.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                tx_log.address == steth_evm_address and
+                hex_or_bytes_to_address(tx_log.topics[1]) == user_address and  # from
+                hex_or_bytes_to_address(tx_log.topics[2]) == wsteth_evm_address  # to # wsteth
+            ):
+                sth_wrapped_amount = asset_normalized_value(
+                    amount=hex_or_bytes_to_int(tx_log.data[:32]),
+                    asset=A_STETH.resolve_to_crypto_asset(),
+                )
+            elif (
+                tx_log.log_index == expected_wsteth_transfer_index and
+                tx_log.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                tx_log.address == wsteth_evm_address and
+                hex_or_bytes_to_address(tx_log.topics[1]) == ZERO_ADDRESS and  # from
+                hex_or_bytes_to_address(tx_log.topics[2]) == user_address  # to # wsteth
```

Comment:
like I told you before. There is no need to search the logs to see this. If it's not available when you are decoding this particular event it will be a bit later. This is what we have action items for. Study how they are used.